cmd/clusterctl/pkg/client/repository/repository_local_unix_test.go

```diff
@@ -0,0 +1,388 @@
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r, err := newLocalRepository(tt.fields.provider, tt.fields.configVariablesClient)
+			if err != nil {
+				t.Errorf("unexpected error: %v", err)
+				return
+			}
+			got, err := r.GetFile(tt.args.version, tt.args.fileName)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if tt.wantErr {
+				return
+			}
+			if string(got) != tt.want.contents {
+				t.Errorf("got %s expected %s", got, tt.want.contents)
+			}
+		})
+	}
+}
+
+func Test_localRepository_GetVersions(t *testing.T) {
+	tmpDir := tempDir(t)
+	defer os.RemoveAll(filepath.Dir(tmpDir))
+
+	// Provider 1: has a single release available
+	dst1 := tempTestLayout(t, tmpDir, "provider-1/v1.0.0/infrastructure-components.yaml", "foo: bar")
+	p1 := config.NewProvider("provider-1", dst1, clusterctlv1.BootstrapProviderType)
+
+	// Provider 2: Has multiple releases available
+	tempTestLayout(t, tmpDir, "provider-2/v1.0.0/infrastructure-components.yaml", "version: v1.0.0")
+	tempTestLayout(t, tmpDir, "provider-2/v1.0.1/infrastructure-components.yaml", "version: v1.0.1")
+	tempTestLayout(t, tmpDir, "provider-2/v2.0.1/infrastructure-components.yaml", "version: v2.0.1")
+	tempTestLayout(t, tmpDir, "provider-2/Foo.Bar/infrastructure-components.yaml", "version: Foo.Bar")
+	tempTestLayout(t, tmpDir, "provider-2/foo.file", "foo: bar")
+	p2URLLatest := "provider-2/latest/infrastructure-components.yaml"
+	p2URLLatestAbs := filepath.Join(tmpDir, p2URLLatest)
+	p2 := config.NewProvider("provider-2", p2URLLatestAbs, clusterctlv1.BootstrapProviderType)
+
+	type fields struct {
+		provider              config.Provider
+		configVariablesClient config.VariablesClient
+	}
+	type want struct {
+		versions []string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		want    want
+		wantErr bool
+	}{
+		{
+			name: "Get the only release available from release directory",
+			fields: fields{
+				provider:              p1,
+				configVariablesClient: test.NewFakeVariableClient(),
+			},
+			want: want{
+				versions: []string{"v1.0.0"},
+			},
+			wantErr: false,
+		},
+		{
+			name: "Get all valid releases available from release directory",
+			fields: fields{
+				provider:              p2,
+				configVariablesClient: test.NewFakeVariableClient(),
+			},
+			want: want{
+				versions: []string{"v1.0.0", "v1.0.1", "v2.0.1"},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r, err := newLocalRepository(tt.fields.provider, tt.fields.configVariablesClient)
+			if err != nil {
+				t.Errorf("unexpected error: %v", err)
+				return
+			}
+			got, err := r.GetVersions()
+			if (err != nil) != tt.wantErr {
+				t.Errorf("error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if tt.wantErr {
+				return
+			}
+			sort.Strings(tt.want.versions)
```

Comment:
Do a length check first to make sure the lengths are the same, and call `t.Fatalf` if they're not